File: src/main/java/gargoyle/ct/convert/impl/CharConverter.java
package gargoyle.ct.convert.impl;

import gargoyle.ct.convert.Converter;
import org.jetbrains.annotations.NotNull;

public class CharConverter implements Converter<Character> {
    @Override
    public String format(Character data) {
        return String.valueOf(data);
    }

    @Override
    public Character parse(@NotNull String data) {
        return data.charAt(0);
    }
}

File: src/main/kotlin/gargoyle/ct/convert/impl/CharConverter.kt
package gargoyle.ct.convert.impl

import gargoyle.ct.convert.Converter

class CharConverter : Converter<Char> {
    override fun format(data: Char): String {
        return data.toString()
    }

    override fun parse(data: String): Char {
        return data[0]
    }
}
